tuplex/codegen/src/LLVMEnvironment.cc

```diff
@@ -1758,5 +1887,129 @@ namespace tuplex {
             return builder.CreateFCmpOLT(fabs_value, eps);
         }
 
+        llvm::BlockAddress* LLVMEnvironment::getUpdateIteratorIndexFunction(llvm::IRBuilder<> &builder, const python::Type &iterableType) {
```

Comment:
maybe rename into generateUpdateIteratorIndexFunction or createUpdateIteratorIndexFunction?

Is an llvm::Function generated? or is function referring to the python function?

Reply:
An llvm::Function is generated and one of its BlockAddress is returned. It it's already generated then just return the BlockAddress from map.
Renamed to createOrGetUpdateIteratorIndexFunctionDefaultBlockAddress.